src/open_inwoner/ssd/xml.py

```diff
@@ -22,18 +22,22 @@ def format_float_repr(value: str) -> str:
 
 
 def format_address(street_name: str, house_nr: str, house_letter: str) -> str:
-    if house_letter:
-        return f"{street_name} {house_nr} {house_letter}"
-    return f"{street_name} {house_nr}"
+    elements = (street_name, house_nr, house_letter)
+    return " ".join((item.strip() for item in elements if item != ""))
 
 
-def format_date(date_str) -> str:
+def format_date(date_str: str) -> str:
+    if not date_str:
+        return date_str
```

Comment:
> the glom that returns the date_str returns "" by default, so this is covered.

The glom happens outside this function so that dependency needs to be remembered. It is much safer to encapsulate and make every part behave correctly. Eg: if the input is not useful return an empty string instead of whatever was put in.

Reply:
The `calculate_loon` and `format_float_repr` also need some of this: both parse the possibly empty string without catching value error:

```py
ValueError: could not convert string to float: ''
ValueError: invalid literal for int() with base 10: ''
```